app/_src/gateway/kong-enterprise/secrets-management/backends/azure-key-vaults.md

```diff
@@ -0,0 +1,174 @@
+---
+title: Azure Key Vaults
+badge: enterprise
+content-type: how-to
+---
+
+The current version of {{site.base_gateway}}'s implementation supports
+configuring
+[Azure Key Vaults](https://azure.microsoft.com/en-us/products/key-vault) in two
+ways:
+
+* Environment variables
+* Managed Identity Authentication
+
+## Configure Azure Key Vaults
+
+{{site.base_gateway}} uses the key to automatically authenticate
+with the Azure Key Vaults API and grant you access.
+
+{:.note}
+> With `Instance Managed Identity Token`, setting the environment variables isn't necessary.
+
+
+Note that Azure AD tenant ID (tenant_id), client ID (client_id), and client secret (client_secret) need to be specified. You can configure these values with environment variables before starting {{site.base_gateway}}:
+
+```bash
+export KONG_VAULT_AZURE_TENANT_ID=tenant_id
+export KONG_VAULT_AZURE_CLIENT_ID=client_id
+export KONG_VAULT_AZURE_CLIENT_SECRET=client_secret
+```
+
+Please note that `client_secret` can only be configured via environments variables.
+
+
+### Examples
+
+Note that Azure's Key Vault support three different secret types.
+
+- Keys
+- Secrets
+- Certificates
+
+For now, Kong only supports the `Secrets` type.
+
+To use a [Secret](https://learn.microsoft.com/en-us/azure/key-vault/general/about-keys-secrets-certificates)
+with the name `secret-name`, create a JSON object in Azure Key Vault that contains one or more properties:
+
+```json
+{
+  "foo": "bar",
+  "snip": "snap"
+}
+```
+
+Note that Azure AD tenant ID (tenant_id), client ID (client_id), and client secret (client_secret) need to be specified. You can configure these values with environment variables before starting {{site.base_gateway}}:
+
+```bash
+export KONG_VAULT_AZURE_TENANT_ID=tenant_id
+export KONG_VAULT_AZURE_CLIENT_ID=client_id
+export KONG_VAULT_AZURE_CLIENT_SECRET=client_secret
+```
+
+```bash
+{vault://azure/secret-name/foo}
```

Comment:
Is the `secret-name` required? I think no as kong will get it from the URI variable
`{vault://azure/foo}`

Reply:
if you check the above examples, we create a secret called `secret-name` which contains.

``` json
{
  "foo": "bar",
  "snip": "snap"
}
```

if you want to query `foo` you have to use. 

`{vault://azure/secret-name/foo}`

for `snip` you need to do 

`{vault://azure/secret-name/snip}`

Reply:
If you have only a single, non-json value as your secret you can do 

`{vault://azure/secret-name}`